apps/dashboard/src/app/api/engine-status/error-rate/route.ts

```diff
@@ -0,0 +1,73 @@
+import { client, v2 } from "@datadog/datadog-api-client";
+import { NextResponse } from "next/server";
+
+export const POST = async () => {
+  const configurationOpts = {
+    authMethods: {
+      apiKeyAuth: process.env.DD_API_KEY,
+      appKeyAuth: process.env.DD_APP_KEY,
+    },
+  };
+  const configuration = client.createConfiguration(configurationOpts);
+
+  configuration.unstableOperations["v2.queryTimeseriesData"] = true;
+  const apiInstance = new v2.MetricsApi(configuration);
+
+  const params: v2.MetricsApiQueryTimeseriesDataRequest = {
+    body: {
+      data: {
+        attributes: {
+          from: new Date().getTime() - 1000 * 60 * 60 * 24,
+          interval: 1000 * 60 * 60,
+          queries: [
+            {
+              dataSource: "metrics",
+              query:
+                "sum:trace.http.request.hits{service:npclabs-engine}.as_count()",
+              name: "success",
+            },
+            {
+              dataSource: "metrics",
+              query:
+                "sum:trace.http.request.errors{service:npclabs-engine}.as_count()",
+              name: "failures",
+            },
+          ],
+          to: new Date().getTime(),
+        },
+        type: "timeseries_request",
+      },
+    },
+  };
+
+  const queryData = await apiInstance.queryTimeseriesData(params);
+
+  // Extract the necessary data
+  const times = queryData?.data?.attributes?.times;
+  const values = queryData?.data?.attributes?.values;
+
+  // Create a map of groupTags to their index in the values array
+  const tagIndexMap: { [tag: string]: number } = {
+    success: 0,
+    failures: 1,
+  };
+
+  const formatedData = times?.map((time, timeIndex) => {
+    const entry: { [key: string]: string | number | undefined } = {
+      time: time,
+    };
+    for (const tag in tagIndexMap) {
+      const value = values?.[tagIndexMap[tag]][timeIndex];
+      entry[tag] = value ? value : 0;
+    }
+    return entry;
+  });
+
+  return NextResponse.json(
+    {
+      data: formatedData,
+      tags: ["failures", "success"],
+    },
+    { status: 200, headers: { "Access-Control-Allow-Origin": "*" } },
```

Comment:
this shouldn't need the CORS header since it will be queried from same domain. is there some other service that will call this outside of the dashboard itself?

Reply:
For some reason I wasn't able to query the endpoint from my localhost. Removed ✅